Controller/AdminResettingController.php

```diff
@@ -11,143 +11,178 @@
         }
 
-        $form = $this->container->get('fos_user.resetting.form');
-        $formHandler = $this->container->get('fos_user.resetting.form.handler');
-        $process = $formHandler->process($user);
+        $form = $formFactory->createForm();
+        $form->setData($user);
+
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $user->setConfirmationToken(null);
+            $user->setPasswordRequestedAt(null);
+            $user->setEnabled(true);
+
+            $message = $this->get('translator')->trans('resetting.flash.success', array(), 'FOSUserBundle');
+            $this->addFlash('success', $message);
+            $response = new RedirectResponse($this->generateUrl('sonata_admin_dashboard'));
 
-        if ($process) {
-            $this->setFlash('fos_user_success', 'resetting.flash.success');
-            $response = new RedirectResponse($this->container->get('router')->generate('sonata_admin_dashboard'));
-            $this->authenticateUser($user, $response);
+            try {
+                $loginManager->logInUser($firewallName, $user, $response);
+                $user->setLastLogin(new \DateTime());
+            } catch (AccountStatusException $ex) {
+                // We simply do not authenticate users which do not pass the user
+                // checker (not enabled, expired, etc.).
+            }
+
+            $userManager->updateUser($user);
 
             return $response;
         }
 
-        return $this->container->get('templating')->renderResponse('SonataUserBundle:Admin:Security/Resetting/reset.html.'.$this->getEngine(), array(
+        return $this->render('SonataUserBundle:Admin:Security/Resetting/reset.html.twig', array(
             'token' => $token,
             'form' => $form->createView(),
-            'base_template' => $this->container->get('sonata.admin.pool')->getTemplate('layout'),
-            'admin_pool' => $this->container->get('sonata.admin.pool'),
+            'base_template' => $this->get('sonata.admin.pool')->getTemplate('layout'),
+            'admin_pool' => $this->get('sonata.admin.pool'),
         ));
     }
+
+    /**
+     * Send an email to a user to confirm the password reset.
+     *
+     * @param UserInterface $user
+     */
+    private function sendResettingEmailMessage(UserInterface $user)
+    {
+        $url = $this->generateUrl('sonata_user_admin_resetting_reset', array(
+            'token' => $user->getConfirmationToken(),
+        ), UrlGeneratorInterface::ABSOLUTE_URL);
+
+        $rendered = $this->renderView($this->container->getParameter('fos_user.resetting.email.template'), array(
+            'user' => $user,
+            'confirmationUrl' => $url,
+        ));
+
+        // Render the email, use the first line as the subject, and the rest as the body
+        $renderedLines = explode("\n", trim($rendered));
```

Comment:
i don't know about ``PHP_EOL`` i currently can't test what kind of line sepatator is used for the email template, and can't that be changed by the user?

but what could be done is:
```php
list($subject, $body) = explode("\n", trim($rendered), 2);
```
but i think that does not work if the email template only has one line because with that it does show "Undefined offset"